From 9d50ae04a11427ee2a764ecbe7167d4f4390ae70 Mon Sep 17 00:00:00 2001
From: Yaoming Qin <b10945@freescale.com>
Date: Fri, 17 Sep 2010 19:32:28 +0800
Subject: [PATCH 078/635] ENGR00131502 Camera: when select zoom button, app
 will crash.100%

  The root cause is that the zoom option is not supported in Camerahal.
  As a result, the java level will call a null obj, which is not initialize
  since the zoom is not supported.

Signed-off-by: Yaoming Qin<B10945@freescale.com>
---
 libcamera/CameraHal.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libcamera/CameraHal.cpp b/libcamera/CameraHal.cpp
index 80f7bbc..94067d5 100755
--- a/libcamera/CameraHal.cpp
+++ b/libcamera/CameraHal.cpp
@@ -191,6 +191,7 @@ void CameraHal::initDefaultParameters()
     strncat( (char*) tmpBuffer, (const char*) CameraParameters::FLASH_MODE_OFF, PARAM_BUFFER);
     p.set(CameraParameters::KEY_SUPPORTED_FLASH_MODES, tmpBuffer);
     p.set(CameraParameters::KEY_FLASH_MODE, CameraParameters::FLASH_MODE_OFF);
+    p.set(CameraParameters::KEY_ZOOM_SUPPORTED, CameraParameters::TRUE);
 
     if (setParameters(p) != NO_ERROR) {
         LOGE("Failed to set default parameters?!");
-- 
1.8.0

